providers/os/resources/npm/package_json.go

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package npm
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+
+	"go.mondoo.com/cnquery/v10/providers-sdk/v1/upstream/mvd"
+)
+
+// packageJson allows parsing the package json file
+type packageJson struct {
+	Name            string                `json:"name"`
+	Description     string                `json:"description"`
+	Version         string                `json:"version"`
+	Private         bool                  `json:"private"`
+	Homepage        string                `json:"homepage"`
+	License         *packageJsonLicense   `json:"license"`
+	Author          *packageJsonPeople    `json:"author"`
+	Contributors    []packageJsonPeople   `json:"contributors"`
+	Dependencies    map[string]string     `jsonn:"dependencies"`
+	DevDependencies map[string]string     `jsonn:"devDependencies"`
+	Repository      packageJsonRepository `json:"repository"`
+	Engines         map[string]string     `jsonn:"engines"`
+	CPU             []string              `json:"cpu"`
+	OS              []string              `json:"os"`
+}
+
+// packageJsonPeople represents the author of the package
+// https://docs.npmjs.com/cli/v10/configuring-npm/package-json#people-fields-author-contributors
+type packageJsonPeople struct {
+	Name  string `json:"name"`
+	Email string `json:"email"`
+	URL   string `json:"url"`
+}
+
+// Explanation:
+//
+//	^: Asserts the start of the string.
+//	([^<]+): Captures one or more characters that are not <. This is for the author's name.
+//	\s+: Matches one or more whitespace characters.
+//	<([^>]+)>: Captures the email address within angle brackets.
+//	\s+: Matches one or more whitespace characters again.
+//	\(([^)]+)\): Captures the URL within parentheses.
+//	$: Asserts the end of the string.
+var authorPatter = regexp.MustCompile(`^([^<]+)\s+<([^>]+)>(?:\s+\(([^)]+)\))?$`)
+
+// UnmarshalJSON implements the json.Unmarshaler interface
+// package.json author can be a string or a structured object
+func (a *packageJsonPeople) UnmarshalJSON(b []byte) error {
+	var authorStr string
+	type authorStruct packageJsonPeople
+	author := authorStruct{}
+
+	// try to unmarshal as structured object
+	err := json.Unmarshal(b, &author)
+	if err == nil {
+		a.Name = author.Name
+		a.Email = author.Email
+		a.URL = author.URL
+		return nil
+	}
+
+	// try to unmarshal as string
+	err = json.Unmarshal(b, &authorStr)
+	if err == nil {
+		matches := authorPatter.FindStringSubmatch(authorStr)
+		if len(matches) == 4 {
```

Comment:
i didnt comb through the regex, but can we not make this only capture the relevant bits? looks like we need 3 matches, not 4 ?

Reply:
1 is the overall match

Reply:
oh, right! makes sense